content/45.user_friendliness.md

```diff
@@ -1,7 +1,67 @@
+By the same token, we have also attempted to provide low-level access to data, and to make as many of those methods accessible and usable as well.
+
+In addition to API considerations, there are three areas that we note have additional care paid to the generation of figures, useful error messages, and to integration with the omnipresent Jupyter environment.
+
 ### Publication-Ready Figures
 
-### Jupyter Integration
+Matplotlib @doi:10.5281/zenodo.7570264 is a fully-featured mechanism for generating figures, with an incredible array of options to customize formatting, appearance, rendering of fonts and glyphs, sleection of colorbars, calculation of tick locations, and appropriate bounds.
+This degree of flexibility provides extremely fine-grained control over the appearance of figures for publication, and Matplotlib provides a large number of output formats for even the most discerning of journals.
+
+yt utilizes matplotlib as its primary rendering engine for visualizations; while 3D renderings (such as those described in @sec:vr) typically generate raw pixel buffers that are saved as images directly, most other visualization functionality in yt relies on matplotlib for generating images that are saved to disk.
+Slices and projections are prepared as variable-resolution images that are pixelized (@sec:pixelization) into fixed resolution buffers, then provided to matplotlib in the `imshow` command.
+Phase plots utilize the `pcolormesh` operation.
+
+All of these internal plots utilize matplotlib's "object-oriented" interface, wherein `Figure` and `Axes` objects are created directly.
+yt then manages the internal state of these objects, and provides high-level access to operations that are aware of the nature of the visualizations; for instance, operations that zoom and pan, and that are unit-aware.
+In the case of 2D spatial images, these are all generated as subclasses of a `PlotWindow` object, named such because it functions as a "window" onto the data, including automatically generating and managing multiple fields simultaneously.
+
+Within the `PlotWindow` object, we set up two levels of data invalidation; when a characteristic of the data is changed (i.e., the viewing window, the resolution) then the *data* is said to be invalidated.
+When a characteristic of the visualization is changed (the label, the scaling of the colorbar, the colormap) then the *plot* is said to be invalidated.
+This allows yt to minimize the overhead of conducting potentially expensive operations on the underlying data, while maximizing its utilization of matplotlib for generation of images.
+The image buffers generated are also always available through the API, so in the not-uncommon case that researchers wish to take the genrated images and perform detailed manipulations or plot layouts, they are able to do so, using the matplotlib API they may already be familiar with.
```

Comment:
```suggestion
The image buffers generated are also always available through the API, so in the common case that researchers wish to take the genrated images and perform detailed manipulations or plot layouts, they are able to do so, using the Matplotlib API they may already be familiar with.
```

Reply:
![not-undelicious](https://user-images.githubusercontent.com/89019/216096287-f8466ca3-f125-48a0-bc7e-c09bd19cd964.jpg)
![food-delicious](https://user-images.githubusercontent.com/89019/216096300-8c7a42ab-1e79-438f-bdbc-5bc59315dc52.jpg)